pkg_resources/__init__.py

```diff
@@ -2958,8 +2959,11 @@ def _find_adapter(registry, ob):
 def ensure_directory(path):
     """Ensure that the parent directory of `path` exists"""
     dirname = os.path.dirname(path)
-    if not os.path.isdir(dirname):
+    try:
```

Comment:
That is why the branch is called "makedirs_exist_ok" :wink: But since the code needs to work in Python < 3.2 I didn't use the new parameter.